crypto/fipsmodule/ec/ec_key.c

```diff
@@ -381,6 +381,32 @@ int EC_KEY_check_fips(const EC_KEY *key) {
     return 0;
   }
 
+  // Check that the coordinates are within the range [0,p-1], when the (raw)
+  // point is affine; i.e. Z=1.
+  // This is the case when validating a received public key.
+  // Note: The check for x and y being negative seems superfluous since
+  // ec_felem_to_bignum() calls BN_bin2bn() which sets the `neg` flag to 0.
+  if(ec_felem_equal(key->pub_key->group, &key->pub_key->group->one, &key->pub_key->raw.Z)) {
```

Comment:
shouldn't the check fail if the coordinates are not affine?

Reply:
We cannot at the moment make this function restricted to affine coordinates without changing how it's used in many tests:
Changing the code to include the Z value check with the rest of the conditions that fail the check
```
    if (!ec_felem_equal(key->pub_key->group, &key->pub_key->group->one, &key->pub_key->raw.Z) ||
        key->pub_key->group->meth->felem_to_bytes == 0 ||
```
crypto_test had the following failures
```
[  FAILED  ] 24 tests, listed below:
[  FAILED  ] All/ECCurveTest.GenerateFIPS/secp521r1, where GetParam() = 16-byte object <CC-02 00-00 00-00 00-00 DC-14 19-B4 FF-FF 00-00>
[  FAILED  ] All/ECCurveTest.GenerateFIPS/secp384r1, where GetParam() = 16-byte object <CB-02 00-00 00-00 00-00 E7-14 19-B4 FF-FF 00-00>
[  FAILED  ] All/ECCurveTest.GenerateFIPS/prime256v1, where GetParam() = 16-byte object <9F-01 00-00 00-00 00-00 F2-14 19-B4 FF-FF 00-00>
[  FAILED  ] All/ECCurveTest.GenerateFIPS/secp224r1, where GetParam() = 16-byte object <C9-02 00-00 00-00 00-00 FD-14 19-B4 FF-FF 00-00>
[  FAILED  ] All/ECDSA_ServiceIndicatorTest.ECDSAKeyCheck/0, where GetParam() = 32-byte object <C9-02 00-00 00-00 00-00 80-B3 41-00 00-00 00-00 01-00 00-00 00-00 00-00 01-00 00-00 00-00 00-00
```
And the ACVP tests failed as well:
```
$ ninja run_tests
[433/437] cd /home/ubuntu/workplace/git-code/aws-lc && /usr...ws-lc/build-fips-shared/testmodulewrapper -tests tests.jso
'ECDSA/keyGen' operation failed.
2021/11/23 19:42:13 failed to process input file: while processing vector set #1: key generation failed for test case 1/1: EOF
2021/11/23 19:42:13 Test failed for "vectors/ECDSA-SigGen.bz2": Failed to process "vectors/ECDSA-SigGen.bz2"
EC_KEY_generate_key_fips failed.
'ECDH/P-224' operation failed.
2021/11/23 19:42:13 failed to process input file: while processing vector set #1: EOF
2021/11/23 19:42:13 Test failed for "vectors/KAS-ECC-SSC.bz2": Failed to process "vectors/KAS-ECC-SSC.bz2"
2021/11/23 19:42:18 Failed 2 tests
exit status 1
```